types/lodash/fp/index.d.ts

```diff
@@ -0,0 +1,618 @@
+         * elements that have a matching property value, else false.
+         *
+         * If an object is provided for predicate the created _.matches style callback returns true for elements that
+         * have the properties of the given object, else false.
+         *
+         * Note: Unlike _.filter, this method mutates array.
+         *
+         * @param predicate The function invoked per iteration.
+         * @return Returns the new array of removed elements.
+         */
+        remove<T>(predicate: Function|Object): (array: List<T>) => T[];
+
+        /**
+         * Removes all provided values from array using SameValueZero for equality comparisons.
+         *
+         * Note: Unlike _.without, this method mutates array.
+         *
+         * @param array The array to modify.
+         * @param values The values to remove.
+         * @return Returns array.
+         */
+        pull<T>(array: T[], ...values: T[]): T[];
+
+        /**
+         * @see _.pull
+         */
+        pull<T>(array: List<T>, ...values: T[]): List<T>;
+
+        keys(arg: any);
+
+        range(min: number, max: number): number;
+
+        /**
+         * Recursively flattens a nested array.
+         *
+         * @param array The array to recursively flatten.
+         * @return Returns the new flattened array.
+         */
+        flattenDeep<T>(array: T): T[];
+
+        /**
+         * Creates an array of elements split into groups the length of size. If collection can’t be split evenly, the
+         * final chunk will be the remaining elements.
+         *
+         * @param size The length of each chunk.
+         * @param array The array to process.
+         * @return Returns the new array containing chunks.
+         */
+        chunk<T>(size: number, array: List<T>): T[][];
+        chunk<T>(size: number): (array: List<T>) => T[][];
+
+        /**
+         * The opposite of _.mapValues; this method creates an object with the same values as object and keys generated
+         * by running each own enumerable property of object through iteratee.
+         *
+         * @param iteratee The function invoked per iteration.
+         * @param object The object to iterate over.
+         * @return Returns the new mapped object.
+         */
+        mapKeys<T>(iteratee, object): Dictionary<T>;
+        mapKeys<T>(iteratee): (object) => Dictionary<T>;
+
+        /**
+         * Checks if value is empty. A value is considered empty unless it’s an arguments object, array, string, or
+         * jQuery-like collection with a length greater than 0 or an object with own enumerable properties.
+         *
+         * @param value The value to inspect.
+         * @return Returns true if value is empty, else false.
+         */
+        isEmpty(value?: any): boolean;
+    }
+}
+
+declare var fp: fp.Static;
+declare module "lodash/fp" {
```

Comment:
just move this into a file fp.d.ts one level higher, and do not declare a global module. look at https://github.com/thalesmello/DefinitelyTyped/blob/22d27bf01182d366a0776ae705d66b836ae308d0/types/lodash/at.d.ts for an example.